src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,131 @@
+package duke;
+
+/**
+ * The Parser class is to make sense of (interpret) the user commend.
+ * Executes different functions based on the user commend/input.
+ */
+public class Parser {
+    /**
+     * Interprets the task from the history file.
+     *
+     * @param userInput Input the previous file history into the application.
+     * @return All the tasks that stored previously in the file.
+     */
+    public static Task parseFile(String userInput) {
+        String recordType = userInput.substring(0, 3);
+        String recordStatus = userInput.substring(3, 6);
+        String recordDescription = userInput.substring(7);
+        Task item = null;
+
+        switch (recordType) {
```

Comment:
Suggest to put these chunks of code into methods inside the execute function in diff task class (etc, todo, event, deadline)